pkg/cmd/accept/exec.go

```diff
@@ -61,95 +64,133 @@ func (o *Options) run() error {
-					csr.Status.Conditions = make([]certificatesv1.CertificateSigningRequestCondition, 0)
-				}
-
-				csr.Status.Conditions = append(csr.Status.Conditions, certificatesv1.CertificateSigningRequestCondition{
-					Status:         corev1.ConditionTrue,
-					Type:           certificatesv1.CertificateApproved,
-					Reason:         fmt.Sprintf("%sApprove", helpers.GetExampleHeader()),
-					Message:        fmt.Sprintf("This CSR was approved by %s certificate approve.", helpers.GetExampleHeader()),
-					LastUpdateTime: metav1.Now(),
-				})
-
-				kubeClient, err := o.ClusteradmFlags.KubectlFactory.KubernetesClientSet()
-				if err != nil {
-					return err
-				}
-				signingRequest := kubeClient.CertificatesV1().CertificateSigningRequests()
-				if _, err := signingRequest.UpdateApproval(context.TODO(), csr.Name, csr, metav1.UpdateOptions{}); err != nil {
-					return err
-				}
+	if csr != nil {
+		if !o.ClusteradmFlags.DryRun {
+			if csr.Status.Conditions == nil {
+				csr.Status.Conditions = make([]certificatesv1.CertificateSigningRequestCondition, 0)
+			}
+
+			csr.Status.Conditions = append(csr.Status.Conditions, certificatesv1.CertificateSigningRequestCondition{
+				Status:         corev1.ConditionTrue,
+				Type:           certificatesv1.CertificateApproved,
+				Reason:         fmt.Sprintf("%sApprove", helpers.GetExampleHeader()),
+				Message:        fmt.Sprintf("This CSR was approved by %s certificate approve.", helpers.GetExampleHeader()),
+				LastUpdateTime: metav1.Now(),
+			})
+
+			kubeClient, err := o.ClusteradmFlags.KubectlFactory.KubernetesClientSet()
+			if err != nil {
+				return false, err
+			}
+			signingRequest := kubeClient.CertificatesV1().CertificateSigningRequests()
+			if _, err := signingRequest.UpdateApproval(context.TODO(), csr.Name, csr, metav1.UpdateOptions{}); err != nil {
+				return false, err
 			}
-			fmt.Printf("CSR %s approved\n", csr.Name)
-		} else {
-			fmt.Printf("no CSR to approve for cluster %s\n", clusterName)
 		}
+		fmt.Printf("CSR %s approved\n", csr.Name)
+		return true, nil
+	}
+	fmt.Printf("no CSR to approve for cluster %s\n", clusterName)
+	return false, nil
+}
 
-		mc, err := clusterClient.ClusterV1().ManagedClusters().Get(context.TODO(),
-			clusterName,
-			metav1.GetOptions{})
-		if err != nil {
-			return err
+func (o *Options) updateManagedCluster(clusterClient *clusterclientset.Clientset, clusterName string) (bool, error) {
+	mc, err := clusterClient.ClusterV1().ManagedClusters().Get(context.TODO(),
+		clusterName,
+		metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
 		}
-		if !mc.Spec.HubAcceptsClient {
-			if !o.ClusteradmFlags.DryRun {
-				mc.Spec.HubAcceptsClient = true
-				_, err = clusterClient.ClusterV1().ManagedClusters().Update(context.TODO(), mc, metav1.UpdateOptions{})
-				if err != nil {
-					return err
-				}
+		return false, err
+	}
+	if !mc.Spec.HubAcceptsClient {
+		if !o.ClusteradmFlags.DryRun {
```

Comment:
ditto

Reply:
same for managed cluste r... we check if the managedcluster is present but in dry-run we don't update it.

Reply:
in that case i think you can do
```go
if o.ClusteradmFlags.DryRun {
  return true, nil
}
if !mc.Spec.HubAcceptsClient {
  ....
}
```

Reply:
yes

Reply:
if I do that, the message `httpAcceptsClient already set for cluster` will be not display in dry-run

Reply:
try this 
```go
if mc.Spec.HubAcceptsClient {
  fmt.Printf("httpAcceptsClient already set for cluster %s\n", clusterName)
  return true, nil
}
if o.ClusteradmFlags.DryRun {
  return true, nil
}

//approve

```